### DISCUSSION TITLE
[Internationalization and localization] Contributors megathread

### DISCUSSION THREAD

**Discussion by 0x7c13** [0x7c13/Notepads#818] [General]
Hey guys,

I am creating this thread to keep track of translators, so people know who is working on what language and for me to keep an record so I can give you guys credits. Whenever I add/update text in Notepads, I will notify you for the updates.

Click [here](https://github.com/JasonStein/Notepads/blob/master/CONTRIBUTING.md) for more info.

Example:

Language: [zh-CN][Simplified Chinese]
Preferred contributor name: Jackie Liu (You can use your Github name as well)
PR Link: <...>

**Comment by 0x7c13** [comment]
Language: [zh-CN][Simplified Chinese]
Preferred contributor name: Jackie Liu

**Comment by frousselet** [comment]
Language: [fr-FR][French (France)]
Preferred contributor name: François Rousselet
PR Link: https://github.com/JasonStein/Notepads/pull/30

**Comment by validatedev** [comment]
Language: [tr-TR][Turkish]
Preferred contributor name: Mert Demir
PR Link: https://github.com/JasonStein/Notepads/pull/38

**Comment by joseppinilla** [comment]
Language: [es-ES][Spanish]
Preferred contributor name: Jose Pinilla
PR Link: #31

**Comment by AmionSky** [comment]
Language: [hu-HU][Hungarian]
Preferred contributor name: Csányi István
PR Link: #44

**Comment by pnp0a03** [comment]
Language: [ja-JP][Japanese]
Preferred contributor name: Mamoru Satoh
PR Link: #52

**Comment by WalterWolf49** [comment]
Language: [de-DE][German]
Language: [de-CH][German (Switzerland)]
Preferred contributor name: Walter Wolf
PR Link: [#48 ](https://github.com/JasonStein/Notepads/pull/48)

**Comment by MairwunNx** [comment]
Language [ru-RU][Russian]
Preferred contributor name: Pavel Erokhin
PR Link: #55

**Comment by sauihdik** [comment]
Language: [fi-FI][Finnish]
Preferred contributor name: Esa Elo
PR Link: #66

**Comment by MairwunNx** [comment]
@JasonStein please, notify here all people if you needed translate update 🥰

**Comment by guari** [comment]
Language: [it-IT][Italian]
Preferred contributor name: Andrea Guarinoni
PR Link: #73

**Comment by tarik02** [comment]
Language [uk-UA][Ukrainian]
Preferred contributor name: [Taras Fomin aka Tarik02](https://github.com/Tarik02)
PR Link: #76

**Comment by 0x7c13** [comment]
@frousselet @validatedev @joseppinilla @AmionSky @pnp0a03 @WalterWolf49 @MairwunNx @sauihdik @guari @Tarik02 
Hey guys, I have added more strings:
1. Tab right click context menu:
https://github.com/JasonStein/Notepads/commit/3da7487c8b0c6949237606f643d0860e69bb0ddc
![image](https://user-images.githubusercontent.com/1166162/61671765-e1399100-ac9d-11e9-98cb-77b80959a515.png)

2. Markdown preview toggle.
https://github.com/JasonStein/Notepads/commit/1fef793e300ffffcb4c0cb63ccfc21a7948e924f
![image](https://user-images.githubusercontent.com/1166162/61671786-f44c6100-ac9d-11e9-84ec-83d89e8846a8.png)

Please take a look and update your translations. Thanks!

**Comment by MairwunNx** [comment]
@JasonStein what base language? for updating strings.

**Comment by 0x7c13** [comment]
> @JasonStein what base language? for updating strings.

en-US

**Comment by 0x7c13** [comment]
@frousselet @validatedev @joseppinilla @AmionSky @pnp0a03 @WalterWolf49 @MairwunNx @sauihdik @guari @Tarik02
Hey guys, I have added more strings:
1.
![image](https://user-images.githubusercontent.com/1166162/62003252-2f4cfb00-b0c9-11e9-884f-2e741ffd3d3b.png)
2.
![image](https://user-images.githubusercontent.com/1166162/62003254-396ef980-b0c9-11e9-8e3f-cddfd6e803f1.png)

**Comment by 0x7c13** [comment]
One more: 
![image](https://user-images.githubusercontent.com/1166162/62004448-21ec3c80-b0da-11e9-8dab-f71ece5aaf5a.png)

**Comment by 0x7c13** [comment]
@frousselet @validatedev @joseppinilla @AmionSky @pnp0a03 @WalterWolf49 @MairwunNx @sauihdik @guari @Tarik02
I have moved all strings in Settings to Settings.resx:https://github.com/JasonStein/Notepads/commit/115aab71a6953ae38a30c0546e4cb8a64496d604

**Comment by JanRajnoha** [comment]
Language: [cs-CZ][Czech]
Preferred contributor name: Jan Rajnoha
PR Link: [#114 ](https://github.com/JasonStein/Notepads/pull/114)

**Comment by 0x7c13** [comment]
Added more strings:

![image](https://user-images.githubusercontent.com/1166162/62755307-40cac700-ba28-11e9-939d-88081f0021e9.png)
![image](https://user-images.githubusercontent.com/1166162/62755320-4d4f1f80-ba28-11e9-8633-b24dd00a8b97.png)
![image](https://user-images.githubusercontent.com/1166162/62755339-6526a380-ba28-11e9-979c-14bbb2be020c.png)

**Comment by pinholucas** [comment]
Language: [pt-BR][Brazilian]
Preferred contributor name: Lucas Pinho B. Santos
PR Link: #154

**Comment by MairwunNx** [comment]
@JasonStein Sorry for the big delay in translating from English to Russian, I'm on vacation now and on about the 24th I can do PR with translation! Thank you for understanding!

**Comment by 0x7c13** [comment]
> @JasonStein Sorry for the big delay in translating from English to Russian, I'm on vacation now and on about the 24th I can do PR with translation! Thank you for understanding!

No worries. Enjoy your vacation.

**Comment by FJduFou** [comment]
Language: [fr-FR][French (France)]
Preferred contributor name: François-Joseph du Fou
PR Link: [#157](https://github.com/JasonStein/Notepads/pull/157) [#158](https://github.com/JasonStein/Notepads/pull/158) [#159](https://github.com/JasonStein/Notepads/pull/159)

**Comment by 0x7c13** [comment]
Some more strings added:  

![image](https://user-images.githubusercontent.com/1166162/64234154-8c975180-ceaa-11e9-9b38-5142ad2137c4.png)
![image](https://user-images.githubusercontent.com/1166162/64234162-93be5f80-ceaa-11e9-970f-9171ba7ba212.png)
![image](https://user-images.githubusercontent.com/1166162/64234682-aedd9f00-ceab-11e9-975c-a652060be3dd.png)
![image](https://user-images.githubusercontent.com/1166162/64241396-84461300-ceb8-11e9-9992-c94ebca66227.png)
![image](https://user-images.githubusercontent.com/1166162/64241790-38e03480-ceb9-11e9-850e-82b1aa160e61.png)

**Comment by 0x7c13** [comment]
Hey guys, I have changed "Shadow Instance" concept to "Shadow Window" to make it easier for users to understand. Please update the translation accordingly. Thanks!

![image](https://user-images.githubusercontent.com/1166162/65025660-fc1a3180-d8eb-11e9-92ca-ad5b8d61a2cd.png)

![image](https://user-images.githubusercontent.com/1166162/65025700-089e8a00-d8ec-11e9-80ab-5bf5ea996fda.png)

A new button was added to launch a new window directly from the menu:

![image](https://user-images.githubusercontent.com/1166162/65025763-24a22b80-d8ec-11e9-9220-a29714ceb1bf.png)

JumpList as well:
![image](https://user-images.githubusercontent.com/1166162/65025615-e6a50780-d8eb-11e9-86fd-90cc9bcde8e5.png)

**Comment by pnp0a03** [comment]
Hi, 
the wording "Shadow Window" is a sort of new word.. for users, it requires some of description about the meaning. I have some of knowledge about UWP multi instance, but hard to get perfectly about the this word :). 
Is it possible to use more generic word - for example, "Secondary Window" "Secondary Instance" or similar one?
best regards,

**Comment by 0x7c13** [comment]
> Hi,
> the wording "Shadow Window" is a sort of new word.. for users, it requires some of description about the meaning. I have some of knowledge about UWP multi instance, but hard to get perfectly about the this word :).
> Is it possible to use more generic word - for example, "Secondary Window" "Secondary Instance" or similar one?
> best regards,

Yes, that is why I remove the text on the indicator and showing an icon instead:
![image](https://user-images.githubusercontent.com/1166162/65068725-006f3a80-d93e-11e9-8604-22d294721e0a.png)

The ONLY place user will see "shadow window" is here:
![image](https://user-images.githubusercontent.com/1166162/65068752-13820a80-d93e-11e9-8a58-2e26ac39af88.png)

It is either "New Window" or "Window" in other places.

**Comment by 0x7c13** [comment]
Added spell checker option and added few strings: https://github.com/JasonStein/Notepads/commit/c46364b75f270bf7129607b388d5ca259f79f6c0

![](https://media.discordapp.net/attachments/597212428635209729/660380643112517643/unknown.png)

It might not work with your locale. Please help verify. If it does not work for your locale, it should fallback to English only. So you can translate to something like this:

> Highlight misspelled words (English only)

**Comment by 0x7c13** [comment]
Added option to set launch preference:
<img width="362" alt="Screen Shot 2020-01-06 at 9 52 37 AM" src="https://user-images.githubusercontent.com/1166162/71837033-4d505b80-306a-11ea-941c-313f3caaca1e.png">
https://github.com/JasonStein/Notepads/commit/bb0770f875248a0712883db345c4e66560dcd949

**Comment by Giandante** [comment]
Hello, I'd like to help with localization of Notepads into sk-SK but I don't really know where to start. Can anyone give me some initial guidance? Thanks!

**Comment by 0x7c13** [comment]
> Hello, I'd like to help with localization of Notepads into sk-SK but I don't really know where to start. Can anyone give me some initial guidance? Thanks!

https://github.com/JasonStein/Notepads/blob/master/CONTRIBUTING.md

**Comment by beerisgood** [comment]
> 
> 
> Language: [de-DE][German]
> Language: [de-CH][German (Switzerland)]
> Preferred contributor name: Walter Wolf
> PR Link: [#48 ](https://github.com/JasonStein/Notepads/pull/48)

@WalterWolf49 are you active? Latest Unigram have some english strings.

**Comment by soumyamahunt** [comment]
Language: [hi-IN][Hindi (India)]
Preferred contributor name: Soumya Ranjan Mahunt
PR Link: [#306](https://github.com/JasonStein/Notepads/pull/306)

**Comment by soumyamahunt** [comment]
Language: [or-IN][Odia (India)]
Preferred contributor name: Soumya Ranjan Mahunt
PR Link: [#309](https://github.com/JasonStein/Notepads/pull/309)

**Comment by 0x7c13** [comment]
New Strings added, need your guys help to translate them into your familiar languages.

![image](https://user-images.githubusercontent.com/1166162/75641059-dc519e00-5beb-11ea-893b-6f4432399e6d.png)

![image](https://user-images.githubusercontent.com/1166162/75641074-ea072380-5beb-11ea-97fd-b58a2d5d9341.png)

![image](https://user-images.githubusercontent.com/1166162/75641090-f4c1b880-5beb-11ea-94b4-0048825486c5.png)

![image](https://user-images.githubusercontent.com/1166162/75641103-fe4b2080-5beb-11ea-8e5b-95a24042fde2.png)

![image](https://user-images.githubusercontent.com/1166162/75641114-06a35b80-5bec-11ea-902c-ce3b6162c115.png)

![image](https://user-images.githubusercontent.com/1166162/75641118-0c00a600-5bec-11ea-9ce5-88b87047a032.png)

![image](https://user-images.githubusercontent.com/1166162/75641166-2f2b5580-5bec-11ea-9856-cb2271be0ba9.png)

**Comment by Daxxxis** [comment]
Language: [pl-PL][polish (Poland)]
Preferred contributor name: Damian Czupryn
PR Link: #335

**Comment by 0x7c13** [comment]
@frousselet  @validatedev @joseppinilla @AmionSky @pnp0a03 @WalterWolf49 @MairwunNx @sauihdik @guari @Tarik02 @Daxxxis @soumyamahunt  @tdh8316 @pinholucas @JanRajnoha 

Here are some more strings I added recently:

![image](https://user-images.githubusercontent.com/1166162/76293225-e586de80-626d-11ea-85fc-b32d970a3167.png)

![image](https://user-images.githubusercontent.com/1166162/76293284-02bbad00-626e-11ea-9d10-2892223f0ce9.png)

**Comment by 0x7c13** [comment]
Auto guess encoding feature has been added:

![image](https://user-images.githubusercontent.com/1166162/76699675-18b1df00-666d-11ea-8f36-cfcbe27dda5f.png)
![image](https://user-images.githubusercontent.com/1166162/76700708-7cd9a080-6677-11ea-9a80-df5eac27f565.png)
![image](https://user-images.githubusercontent.com/1166162/76700713-87943580-6677-11ea-9d6b-c427462dd566.png)

**Comment by szlatkow** [comment]
We wuld like to commit another language on the program , and we were asking how can we test the new language after we make the translate ?

**Comment by 0x7c13** [comment]
Hey guys, there are some newly added strings, please help translate when you get a chance. Thanks!

![image](https://user-images.githubusercontent.com/1166162/77981164-02875e00-72be-11ea-86a4-bc0282a2f9b8.png)
![image](https://user-images.githubusercontent.com/1166162/77981230-2ba7ee80-72be-11ea-933c-da696e779bca.png)
![image](https://user-images.githubusercontent.com/1166162/77981257-3febeb80-72be-11ea-88ca-c4dc7165a99b.png)

**Comment by RadostinMihaylov** [comment]
Could you please give us a hint how to add a bar in the tools window so you could change the language while being in the work window.

**Comment by 0x7c13** [comment]
I am always using Windows language settings to switch between languages:

![image](https://user-images.githubusercontent.com/1166162/78170971-91a08d00-7408-11ea-84e0-f2ddaa8005b5.png)

It asks you to sign out but you can completely ignore it. After switching the language, reopen Notepads and it will use the local that you just switched to.

**Comment by RadostinMihaylov** [comment]
Thank you

На ср, 1 апр 2020 г., 9:05 сл.об. Jiaqi Liu <notifications@github.com>
написа:

**Comment by 0x7c13** [comment]
Hey guys, I have updated some description around Acrylic background settings, please help update the translation accordingly. Thanks!

![image](https://user-images.githubusercontent.com/1166162/79082438-da92f400-7cda-11ea-9dc7-c30dccd4f13e.png)

**Comment by 0x7c13** [comment]
Hi folks,

Added few more strings and need your help here:

![image](https://user-images.githubusercontent.com/1166162/80912458-dd896f00-8cf1-11ea-8976-8771a1e909a0.png)

**Comment by 0x7c13** [comment]
File type display names have been moved to a resource file:

![image](https://user-images.githubusercontent.com/1166162/80940169-c438ff00-8d93-11ea-8def-33e0bba8c860.png)

Please help translate, thanks! 👍

**Comment by JanRajnoha** [comment]
Updated cs-CZ translation

**Comment by jjyao88** [comment]
Language: [zh-TW][Traditional Chinese]
Preferred contributor name: Tony Yao
PR Link: #500

**Comment by 0x7c13** [comment]
Added font style and weight settings, please help translate accordingly:


![image](https://user-images.githubusercontent.com/1166162/81400694-9381fe00-90e2-11ea-81ec-61a5ecbfa04c.png)

**Comment by jjyao88** [comment]
![image](https://user-images.githubusercontent.com/11062997/81403925-05039000-9167-11ea-96ff-9e6e05ccaccf.png)
@JasonStein the font styles should be able to be localized, too

**Comment by 0x7c13** [comment]
> ![image](https://user-images.githubusercontent.com/11062997/81403925-05039000-9167-11ea-96ff-9e6e05ccaccf.png)
> @JasonStein the font styles should be able to be localized, too

Yeah, I will do it as next step. Btw, the appearance of the style and weight is very intuitive. So the need for translation is not that big.

**Comment by 0x7c13** [comment]
Font styles and weights are moved to Resource.resw:
(**You do not have to translate these** unless there are well established translations available in your own language)
![image](https://user-images.githubusercontent.com/1166162/81495685-e84e8180-9266-11ea-8b04-15313561441a.png)

Also added .lrc file supports:
![image](https://user-images.githubusercontent.com/1166162/81495694-f43a4380-9266-11ea-888f-cb6c88fc00a1.png)

**Comment by 0x7c13** [comment]
Added "Rename" feature:
![image](https://user-images.githubusercontent.com/1166162/82048330-59c47080-9669-11ea-8770-c2df01731f99.png)
![image](https://user-images.githubusercontent.com/1166162/82048342-5df08e00-9669-11ea-91f5-8478240fdb3f.png)
![image](https://user-images.githubusercontent.com/1166162/82048356-647f0580-9669-11ea-816c-c7e418126532.png)
![image](https://user-images.githubusercontent.com/1166162/82048362-6779f600-9669-11ea-993e-167c46706367.png)
![image](https://user-images.githubusercontent.com/1166162/82048389-71035e00-9669-11ea-9f48-55b79d1720c7.png)
![image](https://user-images.githubusercontent.com/1166162/82048397-7496e500-9669-11ea-9b56-505b1c300228.png)
![image](https://user-images.githubusercontent.com/1166162/82048409-7a8cc600-9669-11ea-9f41-ebf08d6f4673.png)

**Comment by soumyamahunt** [comment]
Changes have been made to be able to change language in settings. Restart Notepads to complete changes. This would help you all testing changes faster also you can easily contribute for multiple languages.
![50CJlOYwMp](https://user-images.githubusercontent.com/51203900/82114405-4654fa80-977a-11ea-8127-6e0085a167ae.gif)

**Comment by 0x7c13** [comment]
![image](https://user-images.githubusercontent.com/1166162/83296418-9afa6b80-a1a5-11ea-8c34-04cc566b54eb.png)
Added "Smart Copy" feature.

Example:
<before>
![image](https://user-images.githubusercontent.com/1166162/83296527-cbdaa080-a1a5-11ea-8eb9-55be78689f89.png)

<after Ctrl+C or Copy>
![image](https://user-images.githubusercontent.com/1166162/83296546-d1d08180-a1a5-11ea-9639-0668a929d7d9.png)

**Comment by ArmandDelessert** [comment]
Hello
I created [a PR to update the French translations](https://github.com/JasonStein/Notepads/pull/627).

**Comment by HasanAlyazidi** [comment]
Language: [ar-YE][Arabic (Yemen)]
Preferred contributor name: Hasan M Al-Yazidi
PR Link: #772

**Comment by 66Bunz** [comment]
Hi, I added a PR with an update for the italian translation
PR Link: #779

**Comment by bzzrak** [comment]
Language: [sr-Latn-RS][Serbian (Latin)]
Language: [sr-Cyrl-RS][Serbian (Cyrillic)]
Preferred contributor name: bzzrak
PR Link: https://github.com/JasonStein/Notepads/pull/966

**Comment by alessandrocaseti** [comment]
Language: [it-IT][Italian (Italy]
Preferred contributor name: Alessandro
PR Link: https://github.com/JasonStein/Notepads/pull/982

**Comment by 0x7c13** [comment]
<img width="336" alt="Screenshot 2023-01-27 at 8 58 50 PM" src="https://user-images.githubusercontent.com/1166162/215242968-98e59aa6-ed0b-4743-819e-4e3c0acf1f34.png">

**Comment by Stephan-P** [comment]
I have a Dutch translation ready

Language: [nl-NL][Dutch (Netherlands)]
Contributor name: S.Paternotte

Am I actually allowed to create a new language folder and upload the resw files in there?

Edit: correction of language codes

**Reply by 0x7c13** [reply]
Yeah, it is allowed.

**Reply by Stephan-P** [reply]
Hmm, I'm having trouble achieving that.
Could you add a directory "nl-NL" ?

**Reply by Stephan-P** [reply]
[nl-NL.zip](https://github.com/0x7c13/Notepads/files/13847038/nl-NL.zip)
Once these are put in place I can keep them up to date

**Reply by 0x7c13** [reply]
https://github.com/0x7c13/Notepads/pull/1290
